jscomp/super_errors/super_typetexp.ml

```diff
@@ -159,7 +159,11 @@ let setup () =
   Location.register_error_of_exn
     (function
       | Typetexp.Error (loc, env, err) ->
-        Some (Location.error_of_printer loc (report_error env) err)
+        Some (Location.error_of_printer loc (fun ppf ->
```

Comment:
I've tried setting the printer in `Super_location.super_error_reporter` but that's not early enough. The earlier one (with access to file name) would be inside each reporter's `Location.register_error_of_exn`, as shown here.

Maybe I'm missing something? If not, then -bs-reason-error could be the way to go?